freud/box.pyx

```diff
@@ -381,26 +386,18 @@ cdef class Box:
                 Vectors unwrapped by the image indices provided.
         """  # noqa: E501
         vecs = np.asarray(vecs)
-        imgs = np.asarray(imgs)
-
-        if vecs.ndim > 2 or vecs.shape[vecs.ndim-1] != 3:
-            raise ValueError(
-                "Invalid dimensions for vecs given to unwrap. "
-                "Valid input is an array of shape (3,) or (N,3).")
-
-        vecs = freud.common.convert_array(vecs)
+        flatten = vecs.ndim == 1
+        vecs = np.atleast_2d(vecs)
+        vecs = freud.common.convert_array(vecs, shape=(None, 3))
+        imgs = np.atleast_2d(imgs)
         imgs = freud.common.convert_array(imgs, shape=vecs.shape,
                                           dtype=np.int32)
 
-        if vecs.ndim == 1:
-            # only one vector to unwrap
-            vecs = self._unwrap(vecs, imgs)
-        elif vecs.ndim == 2:
-            vecs += imgs[:, [0]]*self.getLatticeVector(0)
-            vecs += imgs[:, [1]]*self.getLatticeVector(1)
-            if self.dimensions == 3:
-                vecs += imgs[:, [2]]*self.getLatticeVector(2)
-        return vecs
+        vecs += imgs[:, [0]]*self.getLatticeVector(0)
+        vecs += imgs[:, [1]]*self.getLatticeVector(1)
+        if self.dimensions == 3:
+            vecs += imgs[:, [2]]*self.getLatticeVector(2)
```

Comment:
Does it work to write `imgs[:, 2]` instead? Avoid putting the single index in brackets if the other form works. Same for other indices above.

Reply:
Using the extra brackets preserves the 2D-ness of the array, which is necessary in this case to multiply each dimension correctly. It subsets an (N, 3) array to an (N, 1) array to multiply the lattice vector, which is of shape (3, ). Without the bracket it would try to multiple an array of shape (N, ) with an array of shape (3, ), which will fail.

Reply:
I actually just removed these lines since it just does what `unwrap` does in C++ side.